R/zchunk_L2232.electricity_FERC_USA.R

```diff
@@ -51,254 +55,489 @@ module_gcam.usa_L2232.electricity_FERC_USA_DISABLED <- function(command, ...) {
+      # All of the supplysector information is the same as before, except the logit exponent
+      A232.structure %>%
+        filter(region == "USA") %>%
+        mutate(logit.year.fillout = min(BASE_YEARS),
+               logit.exponent = subsector.logit,
+               logit.type = subsector.logit.type) %>%
+        select(one_of(LEVEL2_DATA_NAMES[["Supplysector"]])) ->
+        L2232.Supplysector_USAelec
+
+      # L2232.SubsectorShrwtFllt_USAelec: subsector (grid region) share-weights in USA electricity trade
+      # No need to read in subsector logit exponents, which are applied to the technology competition
+      A232.structure %>%
+        filter(region == "USA") %>%
+        select(one_of(LEVEL2_DATA_NAMES[["Subsector"]])) %>%
+        repeat_add_columns(tibble(grid_region = grid_regions)) %>%
+        mutate(subsector = replace(subsector, grepl("grid_region", subsector),
+                                   paste(grid_region[grepl("grid_region", subsector)], "electricity trade", sep = " ")),
+               year.fillout = min(BASE_YEARS),
+               share.weight = 1) %>%
+        select(one_of(LEVEL2_DATA_NAMES[["SubsectorShrwtFllt"]])) ->
+        L2232.SubsectorShrwtFllt_USAelec
+
+      # L2232.SubsectorInterp_USAelec: temporal interpolation of subsector share-weights in USA electricity trade
+      # NOTE: this just carries the base year share-weights forward;
+      # regions that don't export in the base year don't export at all
+      L2232.SubsectorShrwtFllt_USAelec %>%
+        select(one_of(LEVEL2_DATA_NAMES[["Subsector"]])) %>%
+        mutate(apply.to = "share-weight",
+               from.year = max(BASE_YEARS),
+               to.year = max(MODEL_YEARS),
+               interpolation.function = "fixed") ->
+        L2232.SubsectorInterp_USAelec
+
+      # L2232.SubsectorLogit_USAelec: logit exponent of subsector in USA electricity trade
+      # NOTE: There is only one tech per subsector in the FERC markets so the logit choice does not matter
+      L2232.SubsectorShrwtFllt_USAelec %>%
+        mutate(logit.year.fillout = min(BASE_YEARS)) %>%
+        left_join(select(A232.structure, region,
+                         logit.exponent = technology.logit,
+                         logit.type = technology.logit.type),
+                  by = "region") %>%
+        select(one_of(LEVEL2_DATA_NAMES[["SubsectorLogit"]])) ->
+        L2232.SubsectorLogit_USAelec
+
+      # L2232.TechShrwt_USAelec: technology share-weights in USA electricity trade
+      A232.structure %>%
+        filter(region == "USA") %>%
+        select(one_of(LEVEL2_DATA_NAMES[["Tech"]])) %>%
+        repeat_add_columns(tibble(grid_region = grid_regions)) %>%
+        repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+        mutate(subsector = replace(subsector, grepl("grid_region", subsector),
+                                   paste(grid_region[grepl("grid_region", subsector)], "electricity trade", sep = " ")),
```

Comment:
" " is default for `paste` so no real need for it to be specified, but doesn't hurt.